src/python/WMCore/MicroService/Unified/MSManager.py

```diff
@@ -94,17 +94,28 @@ def __init__(self, config=None, logger=None):
 
         # initialize output module
         if 'output' in self.services:
-            self.msOutput = MSOutput(self.msConfig, logger=self.logger)
-            thname = 'MSOutput'
-            self.outputThread = start_new_thread(thname, daemon,
-                                                 (self.output,
-                                                  ['completed',
-                                                   'closed-out',
-                                                   'announced'],
-                                                  self.msConfig['interval'],
-                                                  self.logger))
+            reqStatus = ['completed', 'closed-out', 'announced']
+            self.msOutputConsumer = MSOutput(self.msConfig, logger=self.logger,
+                                             mode='MSOutputConsumer')
+            thname = 'MSOutputConsumer'
+            self.outputConsumerThread = start_new_thread(thname, daemon,
+                                                         (self.outputConsumer,
+                                                          reqStatus,
+                                                          self.msConfig['interval'],
+                                                          self.logger))
+            self.logger.debug(
+                "=== Running %s thread %s", thname, self.outputConsumerThread.running())
+
+            self.msOutputProducer = MSOutput(self.msConfig, logger=self.logger,
+                                             mode='MSOutputProducer')
+            thname = 'MSOutputProducer'
+            self.outputProducerThread = start_new_thread(thname, daemon,
+                                                         (self.outputProducer,
+                                                          reqStatus,
```

Comment:
No it is not! It id not used anywhere. I just fetched it only because I wanted to experiment with updates and synchronization of some dynamic values between Reqmgr and MongoDB (some that are  supposed to change in time). Because all the rest of the values that we are collection right now are static throughout the whole workflow life time. 
We can remove it safely in the future, but for the moment I would like to keep it only to see how it goes and how should we deal with such a synchronization in the future.

Reply:
Almost... :) I am pretty sure they will get aligned at some point for sure. It simply would have taken some time for them to get in sync. And that's what I wanted to test, but anyway, I will leave it for some other time. Removing it from the document for now.